tests/test_mesh.py

```diff
@@ -153,14 +148,86 @@ def test_submesh_meshtags(edim):
             np.arange(num_parent_entities, dtype=np.int32),
             entity_communicator.array.astype(np.int32),
         )
-
         sub_tag, sub_entity_to_parent = scifem.mesh.transfer_meshtags_to_submesh(
             parent_tag, submesh, vertex_to_parent, entity_to_parent
         )
         submesh.topology.create_connectivity(i, edim)
-
         midpoints = dolfinx.mesh.compute_midpoints(submesh, i, sub_tag.indices)
         mesh.topology.create_connectivity(i, mesh.topology.dim)
         parent_midpoints = dolfinx.mesh.compute_midpoints(mesh, i, parent_tag.indices)
 
         np.testing.assert_allclose(midpoints, parent_midpoints[sub_entity_to_parent])
+
+
+@pytest.mark.parametrize("codim", [0, 1, 2])
+@pytest.mark.parametrize("tdim", [1, 2, 3])
+@pytest.mark.parametrize(
+    "ghost_mode", [dolfinx.mesh.GhostMode.none, dolfinx.mesh.GhostMode.shared_facet]
+)
+def test_submesh_creator(codim, tdim, ghost_mode):
+    edim = tdim - codim
```

Comment:
Since you have templated the function (wrt to mesh dtype), it might be good to add a parametrization that test that.

Reply:
What do you mean here? It tests against a DOLFINx Mesh (which indeed can be float32 or float64), but the code itself doesn't really use the dtype for anything but passing the mesh down to the relevant constructors.

Reply:
Ah, OK, I see. Thanks